registrations/utils.py

```diff
@@ -108,37 +108,50 @@ def has_allowed_substitute_user_email_domain(email_address):
     )
 
 
+def _create_calendar_event_from_event(event, time_zone):
+    calendar_event = CalendarEvent()
+
+    if (start_time := event.start_time) and (name := event.name):
+        calendar_event.add("dtstart", start_time.astimezone(time_zone))
+        calendar_event.add("summary", name)
+    else:
+        raise ValueError(
+            "Event doesn't have start_time or name. Ics file cannot be created."
+        )
+
+    end_time = event.end_time if event.end_time else start_time
+    calendar_event.add("dtend", end_time.astimezone(time_zone))
+
+    if description := event.short_description:
+        calendar_event.add("description", description)
+    if location := event.location:
+        location_parts = [
+            location.name,
+            location.street_address,
+            location.address_locality,
+        ]
+        location_text = ", ".join([i for i in location_parts if i])
+        calendar_event["location"] = vText(location_text)
+
+    return calendar_event
+
+
 def create_event_ics_file_content(event, language="fi"):
     cal = Calendar()
     # Some properties are required to be compliant
     cal.add("prodid", "-//linkedevents.hel.fi//NONSGML API//EN")
     cal.add("version", "2.0")
 
-    with translation.override(language):
-        calendar_event = CalendarEvent()
+    local_tz = pytz.timezone(settings.TIME_ZONE)
 
-        if (start_time := event.start_time) and (name := event.name):
-            calendar_event.add("dtstart", start_time)
-            calendar_event.add("summary", name)
+    with translation.override(language):
+        if event.is_recurring_super_event:
+            for sub_event in event.sub_events.filter(deleted=False):
+                calendar_event = _create_calendar_event_from_event(sub_event, local_tz)
+                cal.add_component(calendar_event)
```

Comment:
Should we still generate ICS file if there is at least one valid calender_event?

Admin user can e.g. postpone one of the sub events and in that case the start_time of that particular sub event is set to null. If the read the code correctly the ics file generation will fail in that case.
 
<img width="1324" alt="Screenshot 2024-06-14 at 11 22 41" src="https://github.com/City-of-Helsinki/linkedevents/assets/24706814/d2f973ad-bb07-40c5-b226-147a0072d5c6">

Reply:
How about cancelled events (event_status = "EventCancelled")? Do we want to include those events in the calendar file as well or should they be filtered out.

Reply:
Nice catch 🎣 
Indeed raising a question should the whole series be PUBLISHED or do we just need to care about the published ones. 🤔

Reply:
Nice catch indeed. PO opinion needed here probably so I'll ask aroun in Slack.

Reply:
Changed this to only create the ICS file for sub-events if the sub-events are published, scheduled and not soft-deleted.